libbeat/processors/add_id/config.go

```diff
@@ -0,0 +1,31 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package add_id
+
+// Config for Add ID processor.
+type Config struct {
+	TargetField string `config:"target_field"` // Target field for the ID
+	Type        string `config:"type"`         // Type of ID
+}
```

Comment:
Would be nice if we can Validate the configuration when unpacking. Either by having a custom type for `Type` or by adding `(* Config) Validate() error` method. When we error here, then go-ucfg tells us the full setting name the configuration failed for.